### PR TITLE
Initial + UC 9

### PR DESCRIPTION
**Pull request by Jack-Is-2048**
Initial commit for the Sales Domain

UC 9 is created and will be adequately tested once sales are created

Some design choices were not included in this commit (refer to UML created by Pedro)
> Do we split endpoints for both Sale types?
> > Could mean duplicate code since they include similar actions

> If we don't split endpoints, having 1 Sale controller and service doesn't meet design principles since Sale is abstract
> > Could create abstract controllers and services which are inherited by each type of Sale

### CODE REVIEW

File: src/main/java/com/mercuryCyclists/Sales/controller/StoreController.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+     */
+    @GetMapping
+    public List<Store> getStores() {
+        return storeService.getAllStores();
+    }
+
+    /**
+     * Gets In Store Sales by Store ID
+     */
+    // TODO Test properly
```
Comment (pedrooste) [inline src/main/java/com/mercuryCyclists/Sales/controller/StoreController.java:39]: test?

File: src/main/java/com/mercuryCyclists/Sales/entity/InStoreSale.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * InStoreSale Entity / DAO
+ */
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+@Setter
+@Entity
+@Table(name = "in_store_sale")
+public class InStoreSale extends Sale{
+    // receiptNo should be unique???
```
Comment (pedrooste) [inline src/main/java/com/mercuryCyclists/Sales/entity/InStoreSale.java:19]: good catch, can rm comment

File: src/main/java/com/mercuryCyclists/Sales/entity/OnlineSale.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ * OnlineSale Entity / DAO
+ */
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+@Setter
+@Entity
+@Table(name = "online_sale")
+public class OnlineSale extends Sale {
+    @Column(name = "customer_name")
```
Comment (pedrooste) [inline src/main/java/com/mercuryCyclists/Sales/entity/OnlineSale.java:23]: i think this is automatically implied. Happy to leave

File: src/main/java/com/mercuryCyclists/Sales/controller/StoreController.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    @PostMapping()
+    public Store registerStore(@RequestBody Store store) {
+        return storeService.registerStore(store);
+    }
+
+    /**
+     * Updates existing store based on the store given
+     */
+    @PutMapping()
+    public Store updateStore(@RequestBody Store store) {
```
Comment (pedrooste) [inline src/main/java/com/mercuryCyclists/Sales/controller/StoreController.java:57]: was decided last week that PUTS will take the id in the URI
PUT store/{storeId}

File: src/main/java/com/mercuryCyclists/Sales/controller/StoreController.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        return storeService.getAllStores();
+    }
+
+    /**
+     * Gets In Store Sales by Store ID
+     */
+    // TODO Test properly
+    @GetMapping(path = "/{storeId}/sales")
+    public List<InStoreSale> getStoreSales(@PathVariable("storeId") Long storeId) {
+        return storeService.getStoreSales(storeId);
```
Comment (pedrooste) [inline src/main/java/com/mercuryCyclists/Sales/controller/StoreController.java:42]: I thought we decided to flip this, storing a store from the inStoreSale entity

File: src/main/java/com/mercuryCyclists/Sales/entity/Sale.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+@Setter
+@MappedSuperclass
+public abstract class Sale {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "product_id", nullable = false)
+    private Long productId;
```
Comment (Jack-Is-2048) [inline src/main/java/com/mercuryCyclists/Sales/entity/Sale.java]: Wrong ID, saleId as ID, and productId as normal var

**Review by ryan-a-git — APPROVED** [review]
Will merge for now so I can work on #8.
Jack will address other issues later

**Pull request merged**